File: src/ikc/main/Ioke.Lang/LexicalContext.cs

namespace Ioke.Lang {
    using System.Collections;

    public class LexicalContext : IokeData {
        public object ground;

        public IokeObject surroundingContext;

        public LexicalContext(object ground, IokeObject surroundingContext){
            this.ground = IokeObject.GetRealContext(ground);
            this.surroundingContext = surroundingContext;
        }

        public override string ToString(IokeObject self) {
            return "LexicalContext:" + System.Runtime.CompilerServices.RuntimeHelpers.GetHashCode(self);
        }
    }
}

File: src/ikj/main/ioke/lang/LexicalContext.java
/*
 * See LICENSE file in distribution for copyright and licensing information.
 */
package ioke.lang;

/**
 *
 * @author <a href="mailto:ola.bini@gmail.com">Ola Bini</a>
 */
public class LexicalContext extends IokeData {
    public Object ground;
    public IokeObject surroundingContext;

    public LexicalContext(Object ground, IokeObject surroundingContext) {
        this.ground = IokeObject.getRealContext(ground);
        this.surroundingContext = surroundingContext;
    }

    @Override
    public String toString(IokeObject self) {
        return "LexicalContext:" + System.identityHashCode(self);
    }
}// LexicalContext
